test/Constraints/diagnostics.swift

```diff
@@ -521,7 +521,7 @@ let _ : Color = .rainbow(42)  // expected-error {{argument passed to call that t
 
 let _ : (Int, Float) = (42.0, 12)  // expected-error {{cannot convert value of type '(Double, Float)' to specified type '(Int, Float)'}}
 
-let _ : Color = .rainbow  // expected-error {{member 'rainbow()' is a function; did you mean to call it?}} {{25-25=()}}
+let _ : Color = .rainbow  // expected-error {{function produces expected type 'Color'; did you mean to call it with '()'?}} {{25-25=()}}
```

Comment:
This diagnostic seems strictly better, to me.

Reply:
Looking at the context, it might not be obvious to the reader what function is being referred to here (especially if they don't know you can refer to member functions in an unapplied form like that).

Reply:
Ah, fair point. I’ll see if I can tweak this to be clearer.

Reply:
@varungandhi-apple As of the latest implementation, this diagnostic now reads "Member 'rainbow()' is a function that produces expected type 'Color'; did you mean to call it?" Best of both worlds!